lib/worker/worker.ts

```diff
@@ -0,0 +1,612 @@
+            buffers.push(decryptedData.buffer);
+        }
+
+        return transfer(serialisedResult, buffers);
+
+        // TODO: once we have support for the intendedRecipient verification, we should add the
+        // a `verify(publicKeys)` function to the decryption result, that allows verifying
+        // the decrypted signatures after decryption.
+        // Note: asking the apps to call `verifyMessage` separately is not an option, since
+        // the verification result is to be considered invalid outside of the encryption context if the intended recipient is present, see: https://datatracker.ietf.org/doc/html/draft-ietf-openpgp-crypto-refresh#section-5.2.3.32
+    },
+
+    /**
+     * Backwards-compatible decrypt message function, to be only used for email messages that might be of legacy format.
+     * For all other cases, use `decryptMessage`.
+     */
+    decryptMessageLegacy: async <F extends WorkerDecryptLegacyOptions['format'] = 'utf8'>({
+        decryptionKeys: decryptionKeyRefs = [],
+        verificationKeys: verificationKeyRefs = [],
+        armoredMessage,
+        armoredSignature,
+        binarySignature,
+        ...options
+    }: WorkerDecryptLegacyOptions & { format?: F }) => {
+        const decryptionKeys = await Promise.all(
+            toArray(decryptionKeyRefs).map((keyReference) => keyStore.get(keyReference._idx) as PrivateKey)
+        );
+        const verificationKeys = await Promise.all(
+            toArray(verificationKeyRefs).map((keyReference) => keyStore.get(keyReference._idx))
+        );
+
+        const signature = await getSignatureIfDefined(binarySignature || armoredSignature);
+
+        const {
+            signatures: signatureObjects,
+            ...decryptionResultWithoutSignatures
+        } = await decryptMessageLegacy<Data, F>({
+            ...options,
+            message: armoredMessage,
+            signature,
+            decryptionKeys,
+            verificationKeys
+        });
+
+        const serialisedResult = {
+            ...decryptionResultWithoutSignatures,
+            signatures: signatureObjects.map((sig) => sig.write() as Uint8Array) // no support for streamed input for now
+        };
+
+        const buffers = serialisedResult.signatures.map((sig) => sig.buffer);
+        if (options.format === 'binary' && serialisedResult.data instanceof Uint8Array) { // even with 'format: binary', string data is returned for legacy messages
+            const decryptedData = serialisedResult.data;
+            buffers.push(decryptedData.buffer);
+        }
+
+        return transfer(serialisedResult, buffers);
+    },
+    /**
+     * Generating a session key for the specified symmetric algorithm.
+     * To generate a session key based on some recipient's public key preferences,
+     * use `generateSessionKeyFromKeyPreferences()` instead.
+     */
+    generateSessionKey: async (algoName: Parameters<typeof generateSessionKey>[0]) => {
```

Comment:
Can we swap these? I.e. `generateSessionKey` takes `{ recipientKeys }`, and call the other one `generateSessionKeyForAlgorithm`. As the latter is the more low-level one.